sale_order_country_allowed_product/models/sale_order.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2025 Manuel Regidor <manuel.regidor@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    unavailable_product_msg = fields.Html(compute="_compute_unavailable_product_msg")
+
+    @api.depends("order_line.product_id")
+    def _compute_unavailable_product_msg(self):
+        for order in self:
+            unavailable_product_msg = ""
+            if any(not line.country_available for line in order.order_line):
+                products = order.order_line.filtered(
+                    lambda line: not line.country_available
+                ).mapped("product_id")
+                if products:
+                    products_list = (
+                        f'<ul>{"".join(f"<li>{p.name}</li>" for p in products)}</ul>'
+                    )
+                    unavailable_product_msg = _(
+                        "The following products are not available in the shipping country: "
+                        "%(products_list)s"
+                    ) % {"products_list": products_list}
```

Comment:
Please use the [correct method](https://www.odoo.com/documentation/16.0/developer/howtos/translations.html#variables) for the variable translation.